lib/database_consistency/checkers/enum_checkers/enum_column_checker.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module DatabaseConsistency
+  module Checkers
+    # This class checks that ActiveRecord enum is backed by enum column
+    class EnumColumnChecker < EnumChecker
```

Comment:
Now, this checker intersects with `EnumTypeChecker` with a shorter list of "allowed" types. Before, we were okay with many types, but now we want to promote `enum` types. 

I don't think we should have a single checker as this could make code harder, but (I believe) you would agree that having both checkers enabled simultaneously doesn't make much sense. 

From one side, this looks more like a mode (strict/open). However, the gem doesn't support modes per checkers, and I'm wondering if we should have such (similar to what Rubocop has, for example). 

If we don't go in that direction, maybe we should have an internal toggle system that turns off/on some checkers based on intersections. For example, if this checker is on, then `EnumTypeChecker` is off. However, if we go that way, I'm afraid we may end up with something too complex and unobvious. 

With that said, I tend to have modes but I would like to hear your opinion on this matter.